engine/src/main/java/io/zeebe/engine/processing/streamprocessor/writers/TypedEventWriter.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processing.streamprocessor.writers;
+
+import io.zeebe.msgpack.UnpackedObject;
+import io.zeebe.protocol.impl.record.RecordMetadata;
+import io.zeebe.protocol.record.intent.Intent;
+import java.util.function.Consumer;
+
+public interface TypedEventWriter {
+
+  void appendNewEvent(long key, Intent intent, UnpackedObject value);
+
+  void appendFollowUpEvent(long key, Intent intent, UnpackedObject value);
+
+  void appendFollowUpEvent(
```

Comment:
I am not sure we need this method. I quickly sketched out an experiment where I removed this method, and it looks fine to me. See `6171-review-experiment-pihme`

While doing that I saw that this construct is already used e.g. in  `TypedCommandWriter`. So I understand the desire for consistency.

Then I was trying to figure out which code is actually using this and I could find only one place:
```
          writer.appendFollowUpCommand(
              key, JobIntent.TIME_OUT, record, (m) -> m.valueType(ValueType.JOB));
```
in `JobTimeoutTrigger` 

So in the one case I found where it is used, the consumer is used to modify the meta data. This is a little unexpected to me. Since it is a consumer I thought it would just read the metadata, not modify it. 

So another option to simply removing it could also be to rename the last parameter to `metadataModifier` or something like this, or to change the type of the last parameter to `UnaryOperator<RecordMetadata> metaDataModifier`.

Happy to discuss this 1 on 1. I only had a couple of minutes to look into this, so maybe I missed something.

Reply:
After our discussion and our quick look into this, I read what you said again and I totally agree that it's strange that this is a Consumer. It makes it look like a return value or a callback function, but it's used especially to create a last minute modification to the "consumed" metadata. I agree that `UnaryOperator<RecordMetadata>` describes the intent better. Personally, I would then opt to name it `UnaryOperator<RecordMetadata> modifier`. 

However, I looked into this modification used in `JobTimeoutTrigger`. It actually doesn't achieve anything. The `record` actually is an instance of `JobRecord` which will lead to `ValueType.JOB` in the `TypedStreamWriterImpl`. 😅 

Personally, I'd like to get rid of the whole thing. Perhaps @saig0 can chime in. Do you know any reason to keep this alive still? If you want we can also discuss the topic f2f.

Reply:
If we don't find a usage then we can remove :sweat_smile: I'm always happy to get rid of something that is not needed anymore :smile:

Reply:
We've improved them with `UnaryOperator` named `modifier`, and deprecated these methods